### PR TITLE
Support static import package selective imports

### LINKED ISSUE #663: Support import package selective package and script-module imports
## Motivation

The `{import}` package provides selective imports from installed packages and R script modules:

```r
import::from(Hmisc, impute, nomiss)
import::from(dplyr, keep_when = filter)
import::here(fun_a, fun_b, .from = "helpers.R")
```

It supports aliases, `.all`, `.except`, imports into named search-path entries or local environments, and module files evaluated in private environments.

Upstream documentation:

- https://rticulate.github.io/import/
- https://rticulate.github.io/import/articles/import.html

Without dedicated handling, Raven may flag imported bindings as undefined, miss the module relationship, and cannot navigate imported names back to their package/module definitions.

## Proposed first phase

Recognize statically analyzable `import::from()`, `import::here()`, and `import::into()` forms:

- package imports using Raven's package export metadata;
- literal local `.R` module paths;
- explicit selected names;
- renamed imports;
- `.all = TRUE` and static `.except` when the export/member set is known;
- the documented default and explicitly named `.into` search-path destinations;
- `import::here()` bindings scoped to the calling/module environment;
- module dependencies without leaking unselected/private names.

Bindings should be position-sensitive unless the upstream form provides a stronger file/module-level guarantee. Changes to a local imported module should revalidate importers.

## Non-goals for the first phase

- Executing R.
- Dynamic `.character_only` package/module vectors.
- URL/pins-backed modules.
- Arbitrary computed `.directory`, `.library`, or `.into` environments.
- Complete simulation of attachment/detachment side effects or experimental S3 registration.

Unsupported dynamic forms should fail conservatively and remain compatible with explicit Raven directives.

## Reuse

The implementation should reuse a general selective-import contribution developed for `{box}` where the semantics overlap:

- dependency identity;
- selected and renamed bindings;
- optional alias/search-path binding;
- private module environment;
- interface hashing and dependent revalidation.

The detectors and path-resolution rules must remain package-specific: `{import}` modules are evaluated and cached differently from `{box}` modules.

## Acceptance tests

- package and local-module imports;
- explicit, renamed, `.all`, `.except`, `from`/`here`/`into` forms;
- missing exports/modules, aliases, shadowing, and position gating;
- private module dependencies and nested imports;
- edit/create/delete revalidation, open/closed files, CLI/LSP parity;
- package metadata fallback tiers and no regression to ordinary `library()` or `source()`.

### PR DESCRIPTION
**Pull request by jbearak**
## Summary

- add first-phase static support for `import::from()`, `import::here()`, and `import::into()` using the shared `selective_import` abstraction introduced for `{box}`
- resolve installed-package exports through `PackageLibrary` completeness tiers and literal `.R`/`.r` script modules through an isolated `import_pkg` detector/path/export policy
- model explicit and renamed selections, `.all`, static `.except`, default and named destinations, function/current-environment scope, typed non-lending module dependencies, watcher/replay revalidation, and CLI/LSP parity
- preserve exact package/export identity for renamed bindings across completion, hover, signatures, parameter resolution, and references without leaking unselected package exports or private module members
- retain named search-path destinations as lower-priority fallback layers across recursive, streaming, sourced, parent-prefix, and ordered batch scope

## Static boundary

Deliberately unsupported forms remain conservative/inert: executing R, dynamic `.character_only` vectors, URL/pins-backed modules, arbitrary computed `.directory`/`.library`/`.into` environments, environment-valued destinations, complete attach/detach or S3 side-effect simulation, and `import:::` internal imports.

## Tests

Added focused and integration coverage for:

- package and local script imports, explicit/renamed selections, `.all`, `.except`, destinations, position/function scope, dotted names, privacy, and nested imports
- missing modules, case-only paths, authoritative missing exports, version-qualified packages, absolute script paths, and exact multiline member ranges/navigation
- package/export identity in completion, hover, signatures, parameter resolution, and references, including same-spelling isolation
- local-module edit/create/delete/replay behavior, typed selective-module package warming, ordinary-source-neighborhood warming, and recursive/streaming scope parity
- no regression to `{box}`, ordinary package loads, or ordinary source relationships

Independent gates run locally:

- `cargo fmt --all --check`
- `cargo clippy --workspace --all-targets --features test-support -- -D warnings`
- `RUSTDOCFLAGS="-D warnings" cargo doc -p raven --no-deps --document-private-items`
- `RUSTDOCFLAGS="-D warnings" cargo doc -p raven --no-deps --document-private-items --features test-support`
- `cargo test --workspace --features test-support` — 6,801 library tests passed, 12 ignored; all integration suites and 53 doctests passed
- `git diff --check`

## Documentation

Updated README, cross-file behavior, completion, diagnostics, go-to-definition, find-references, CLI parity, limitations, development architecture, and v0.18.0 release notes.

No pre-existing bugs were found or fixed as part of this issue.

Closes #663

## Summary by CodeRabbit

* **New Features**
  * Expanded static `{import}` support for `import::from()`, `import::here()`, and `import::into()` with package selective bindings, destinations, exclusions, and wildcard behavior.
  * Added import-aware navigation and enrichment across diagnostics, completions, hover/signatures, goto-definition, and references.

* **Bug Fixes**
  * Improved cross-file selective-module dependency tracking, workspace index replay, importer refresh, and package warming for selective imports.
  * Updated `raven check` warming and metadata enrichment to follow `{import}` call behavior.

* **Documentation**
  * Refreshed guides and CLI/diagnostics documentation to reflect package selective imports and new import-related diagnostic codes.

* **Tests**
  * Added/updated end-to-end and regression coverage for selective-import refresh and resolution edge cases.

### CODE REVIEW

File: crates/raven/src/backend.rs [hunk 1]
```diff
@@ -6326,50 +6326,73 @@ pub async fn run_system_file_convergence_for_test(
...
+            crate::cross_file::enrich_selective_import_resolutions(
+                metadata,
+                uri,
+                expansion_root.as_ref(),
+            );
+        }
+        prepared
+    })
+    .await
+    .unwrap_or_default();
```
[the remark this answers is not in the document]
Comment (jbearak) [reply crates/raven/src/backend.rs]: Fixed in 211efc0a. The spawn_blocking join error is now logged before the replay falls back to an empty prepared set.

File: crates/raven/src/box_use/resolve.rs [hunk 2]
```diff
@@ -224,6 +235,43 @@ impl<'a> ArtifactModuleExportEnv<'a> {
...
+                if let Some(provenance) = provenance {
+                    let order = (import.line, import.column);
+                    if best
+                        .as_ref()
+                        .is_none_or(|(best_order, _)| order >= *best_order)
+                    {
+                        best = Some((order, provenance));
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (jbearak) [reply crates/raven/src/box_use/resolve.rs:283]: Fixed in 211efc0a. Recursive export/name expansion now disables eager provenance, and local member provenance follows the selected source explicitly with the active visited/depth guard. Added a mutually importing import-package module regression.

File: crates/raven/src/cli/check.rs [hunk 3]
```diff
@@ -1090,9 +1102,36 @@ fn reported_packages_to_warm(
...
                     },
                 ));
+                packages.extend(
+                    metadata
+                        .import_calls
+                        .iter()
+                        .filter_map(crate::import_pkg::ImportCall::package_name)
+                        .map(str::to_string),
+                );
             }
```
[the remark this answers is not in the document]
Comment (jbearak) [reply crates/raven/src/cli/check.rs:1130]: Fixed in 211efc0a. Chunk metadata is now enriched before warming and its resolved local-module URIs seed the shared selective-module closure. Added an Rmd chunk -> module.R -> package import regression.
